mapper-processor/src/main/java/com/datastax/oss/driver/internal/mapper/processor/dao/DaoMethodGenerator.java

```diff
@@ -42,6 +42,8 @@
   protected final ExecutableElement methodElement;
   protected final DaoImplementationSharedCode enclosingClass;
   protected final ProcessorContext context;
+  protected final String FUNCTION_STATEMENT_BUILDER_TYPE =
+      "java.util.function.Function<com.datastax.oss.driver.api.core.cql.BoundStatementBuilder,com.datastax.oss.driver.api.core.cql.BoundStatementBuilder>";
```

Comment:
I do no like this. I tried for a few hours to come up with a better method for type matching than String comparison and came up empty handed. It's relatively trivial to match A TypeMirror object to a Class provided generics are not involved. I was able to figure out how introspect and check the function generics  parameters against a class. The problem was that the TypeMirror object representing the Function parameter would not match against Function.class, and I was not able to construct a TypeMirror object that included the generics. If anyone has a better way to do this I'm all ears.

Reply:
You need to extract the element from the mirror to match the raw type, and then match the type parameters separately. I think this will do it:
```java
VariableElement parameter = ...
TypeMirror mirror = parameter.asType();
if (mirror.getKind() == TypeKind.DECLARED) {
  DeclaredType declaredType = (DeclaredType) mirror;
  if (context.getClassUtils().isSame(declaredType.asElement(), Function.class)
      && context.getClassUtils().isSame(declaredType.getTypeArguments().get(0), BoundStatementBuilder.class)
      && context.getClassUtils().isSame(declaredType.getTypeArguments().get(1), BoundStatementBuilder.class)) {
    // match
  }
}
```

Reply:
This worked, it occurred to me to use The DeclaredType to match the TypeArguments of the function, but I didn't understand that isSame would would with the element but not with TypeMirror.